Add TypeDoc references (#1685)
LiskHQ/lisk-docs: modules/ROOT/pages/modules/index.adoc

:url_module_mci: {site-url}/lisk-sdk/v6/references/typedoc/classes/lisk_framework.MainchainInteroperabilityModule.html
:url_module_pos: {site-url}/lisk-sdk/v6/references/typedoc/classes/lisk_framework.PoSModule.html
:url_module_sci: {site-url}/lisk-sdk/v6/references/typedoc/classes/lisk_framework.SidechainInteroperabilityModule.html
:url_module_random: {site-url}/lisk-sdk/v6/references/typedoc/classes/lisk_framework.RandomModule.html
:url_module_reward: {site-url}/lisk-sdk/v6/references/typedoc/classes/lisk_framework.RewardModule.html
:url_module_token: {site-url}/lisk-sdk/v6/references/typedoc/classes/lisk_framework.TokenModule.html
:url_module_validators: {site-url}/lisk-sdk/v6/references/typedoc/classes/lisk_framework.ValidatorsModule.html

Overview of all available modules for the Lisk SDK.

[cols="1,3",options="header",stripes="hover"]
|===
|Name
|Description

| {url_module_base}[Base module]
| All other modules must extend from the Base module.

| {url_module_auth}[Auth module]
|Responsible for handling and verifying nonces and for transaction signature validation, including transactions from multi-signature accounts.

| {url_module_fee}[Fee module]

> |Responsible for handling the fee of transactions.

Reviewer: I think we should reword this heading to the following:
Responsible for handling the transaction fees.

It allows chains to choose the token used to pay the fee and to define a minimum fee for transactions to be valid.

| {url_module_mci}[Mainchain Interoperability module]
|Provides basic functionalities for the mainchain to transmit information between interoperable chains in the Lisk ecosystem using cross-chain messages.

| {url_module_pos}[PoS module]

> |Handles all aspects of the generator selection, this includes the registration of accounts as validators, the staking process, and potential reports of misbehavior.

Reviewer: imho, I think we should have either one of the following:
'Handles all aspects of the generator selection, this includes the registration of accounts for validators, the staking process, and potential reports of misbehavior.'

Or alternatively:

'Handles all aspects of the generator selection, this includes the registration of validator accounts, the staking process, and potential reports of misbehavior.'

| {url_module_sci}[Sidechain Interoperability module]
|Provides basic functionalities for a sidechain to transmit information between interoperable chains in the Lisk ecosystem using cross-chain messages.

| {url_module_random}[Random module]
a|Handles the validation of the inputs and computation of outputs for the commit and reveal process of a Lisk blockchain.

| {url_module_reward}[Reward module]
a|Provides the base reward system for a blockchain developed with the Lisk SDK

| {url_module_token}[Token module]

> |Handles minting, burning, and transferring tokens int he Lisk ecosystem.

Author: ```suggestion
|Handles minting, burning, and transferring tokens inside the Lisk ecosystem.
```
